src/train.py

```diff
@@ -199,91 +343,57 @@ def train_speech_auto(args):
             if train_log is not None:
                 train_log.add_scalar("loss", losses[-1], global_step)
             global_step += 1
-
-        # evaluate(model, valid_dataset)
-
         avg_l = np.mean(losses)
         print("epoch %-3d \t loss = %0.3f \t" % (epoch, avg_l))
-        # if validation < best:
-        # TODO: Add model (and optimizer) saving for reloading training
-        #     print("Saving model!")
-        #     best = validation
-        #     model.save_model()
     return model
 
-def train(args):
-    set_seed(args.seed)
+
+def initialize_model(args):
+    """
+        Using args, initialize starting epoch, best per, model, optimizer
+    """
+    if args.load_path is not None:
+        model = UNAST(None, None, None)
+        optimizer = toch.optim.Adam(model.parameters(), lr=args.lr, weight_decay=1e-5)
+        s_epoch, best, model, optimizer = load_ckp(args.load_path, model, optimizer)
+        s_epoch = s_epoch + 1
+    else:
+        text_m, speech_m, discriminator = None, None, None
+        if args.model_type == 'rnn':
+            text_m = TextRNN(args).to(DEVICE)
+            speech_m = SpeechRNN(args).to(DEVICE)
+        elif args.model_type == 'transformer':
+            # TODO: Fill it in
+            pass
+
+        if args.use_discriminator:
+            # TODO: Fill it in
+            pass
+        model = UNAST(text_m, speech_m, discriminator)
+
+        # initialize optimizer
+        optimizer = None
+        if args.optim_type == 'adam':
+            optimizer = torch.optim.Adam(model.parameters(), lr=args.lr, weight_decay=1e-5)
+            s_epoch, best = 0, 100
+
+    return s_epoch, best, model, optimizer
```

Comment:
I like the new function, and this is fine for now. We'll define these hyperparameters later.